common/src/main/java/org/opensearch/ml/common/transport/batch/MLBatchIngestionInput.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.common.transport.batch;
+
+import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import lombok.Builder;
+import lombok.Getter;
+
+/**
+ * ML batch ingestion data: index, field mapping and input and out files.
+ */
+public class MLBatchIngestionInput implements ToXContentObject, Writeable {
+
+    public static final String INDEX_NAME_FIELD = "index_name";
+    public static final String FIELD_MAP_FIELD = "field_map";
+    public static final String DATA_SOURCE_FIELD = "data_source";
+    public static final String CONNECTOR_CREDENTIAL_FIELD = "credential";
+    @Getter
+    private String indexName;
+    @Getter
+    private Map<String, Object> fieldMapping;
+    @Getter
+    private Map<String, String> dataSources;
+    @Getter
+    private Map<String, String> credential;
+
+    @Builder(toBuilder = true)
+    public MLBatchIngestionInput(
+        String indexName,
+        Map<String, Object> fieldMapping,
+        Map<String, String> dataSources,
+        Map<String, String> credential
+    ) {
+        if (indexName == null) {
+            throw new IllegalArgumentException("index name for ingestion is null");
+        }
+        if (dataSources == null) {
+            throw new IllegalArgumentException("dataSources for ingestion is null");
+        }
+        this.indexName = indexName;
+        this.fieldMapping = fieldMapping;
```

Comment:
Is `fieldMapping` optional ?

Reply:
Yes, it's not listed as required here because it could be used for general ingestion for any file without LLM input/output mapping needed.